File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Python/s185737640_comments_removed.py
def is_even_string(s):
    mid = len(s) // 2
    return s[:mid] == s[mid:]

def main():
    S = input().strip()

    if len(S) % 2 == 1:
        S = S[:-1]
    else:
        S = S[:-2]

    while S and not is_even_string(S):
        S = S[:-2]

    print(len(S))

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Go/s185737640_comments_removed.go
package main

import (
	"fmt"
)

func main() {
	var s string
	fmt.Scanln(&s)

	if len(s)%2 == 1 {
		s = s[:len(s)-1]
	} else {
		s = s[:len(s)-2]
	}

	for len(s) > 0 {
		if isEvenString(s) {
			break
		}
		s = s[:len(s)-2]
	}
	fmt.Println(len(s))
}

func isEvenString(s string) bool {
	a := s[:len(s)/2]
	b := s[len(s)/2:]
	return a == b
}

// End of Code
